packages/next/next-server/lib/router/router.ts

```diff
@@ -1112,7 +1112,12 @@ export default class Router implements BaseRouter {
               pages
             )
 
-            if (pages.includes(parsedHref.pathname)) {
+            const isDestinationAKnownPage = pages.includes(parsedHref.pathname)
+            const isDestinationARewrittenUrl = rewrites.afterFiles.some(
+              (x: any) => x.source === parsedHref.pathname
+            )
+
+            if (isDestinationAKnownPage || isDestinationARewrittenUrl) {
```

Comment:
It looks like we can simplify this check to instead check if the destination starts with `/` and if it does allow `this.change` to handle the transition/redirect since it resolves rewrites/pages already

Reply:
So I spent some time thinking of your suggestion and I am not really sure we can implement it : it seams like if the destionation starts with `/` but this destionation isn't known by next, then the fallback method won't be applied (replace the href using the browser's API `window.location.href = destination`) and thus leading to potential regression. WDYT ?

Reply:
If a page/rewrite isn't handled in `this.change` we fallback to `window.location.href` already so should work as expected.

Reply:
Indeed you were right, it seems to work ! 

On a side note, do you know why some pipelines are working and some aren't ? I'm having trouble finding if the failed pipeline are because of changes in this PR or something else. WDYT ?

Reply:
The Azure tests can be ignored currently as they flake from the environment currently, I had to allow the GitHub actions CI step to run since this is your first PR, that is the one we focus on currently